nipoppy/workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -489,23 +491,25 @@ def run(participant_id, global_configs, session_id, output_dir, use_bids_filter,
             ## check that vectors are unique
             tvec = bvec[:,tndir]
             tdir = np.unique(tvec, axis=0)
-
-            ## compute and print the maximum shell - can get odd numbers?
-            tlmax = int(np.floor((-3 + np.sqrt(1 + 8 * tdir.shape[1])) / 2.0))
+            mldir.append(tdir.shape[1])
+            
+            ## compute and print the maximum shell
+            tlmax = int(np.floor((-3 + np.sqrt(1 + 8 * tdir.shape[1]) / 2.0)))
             mlmax.append(tlmax)
             logger.info(f" -- Shell {int(shell)} has {tdir.shape[1]} directions capable of a max lmax: {tlmax}")
 
-        ## the minimum lmax across shells is used
-        plmax = min(mlmax)
-        logger.info(f"The maximum lmax across shells is: {plmax}")
+        ## the max lmax within any 1 shell is used
+        plmax = max(mlmax)
+        logger.info(f"The maximum lmax for any one shell is: {plmax}")
         
     ## if lmax too large, reset with warning
     if sh_order <= plmax:
         logger.info(f"Running model with requested lmax: {sh_order}")
     else:
         logger.info(f"The requested lmax ({sh_order}) exceeds the recommended capabilities of the data ({plmax})")
+        logger.info(f"Generally, you do not want to fit an lmax in excess of any one shell's ability in the data.")
         logger.info(f" -- You should redo with sh_order set to: {plmax}")
-        logger.info(f"Running model with overrode lmax: {sh_order}")
+        #logger.info(f"Running model with overrode lmax: {sh_order}")
         #sh_order = plmax
         
     ## hard coded inputs to the tractoflow command in mrproc
```

Comment:
Are these `global vars` or `run args`? Would be nice to either move them to the top or include in the argparse?

Reply:
`sh_order` is a `run_args` passed by the user that has a default in the run script. We could make the default adjustable in global, but that might be more trouble than it's worth. 

`plmax` has to be determined by the data - it's something the user _can_ figure out beforehand so they can set it correctly, but in my experience it's worth parsing and checking this upfront.

What I'm doing with these steps is verifying the data have enough directions to fit that parameter, otherwise it will fail once it starts running.